tracer/src/Datadog.Trace/ContinuousProfiler/ContextTracker.cs

```
@@ -85,11 +85,21 @@ public void Reset()
             WriteToNative(SpanContext.Zero);
         }
 
-        private void EnsureIsInitialized()
+        private bool EnsureIsInitialized()
         {
-            if (_traceContextPtr.IsValueCreated)
+            try
             {
-                return;
+                if (_traceContextPtr.IsValueCreated)
+                {
+                    return true;
+                }
+            }
+            catch (ObjectDisposedException e)
```

why not a `pokemon cactch` ? (catch all exceptions instead)

---

✅ done